### DISCUSSION TITLE
hideCursor not working

### DISCUSSION THREAD

**Discussion by redimongo** [katspaugh/wavesurfer.js#2764] [Q&A]
I have tried everything, but I can't get hideCursor function to work I am using the latest `https://unpkg.com/wavesurfer.js`

and I have also added `https://unpkg.com/wavesurfer.js/dist/plugin/wavesurfer.cursor.js`

I am wondering what is the right way to hideCursor please.

```
var wavesurfer = WaveSurfer.create({
                    container: waveformContainer,
                    hideCursor:true,
                    waveColor: '#7bd961',
                    progressColor: '#5ca740',
                    cursorColor: '#7bd961',
                    barWidth: 4,
                    barHeight: 0.3,
                    barRadius: 4,
                    pixelRatio: 2,
                    partialRender: true,
                    plugins: [
                        WaveSurfer.cursor.create({
                            showTime: false, // Hide the time display
                            hideCursor: true // Hide the cursor
                        })
                    ]
                });
```

**Comment by katspaugh** [comment]
I think we need to rename some of the "cursor" options because they all refer to different cursors which is confusing:

* `hideCursor` – this one hides the mouse pointer, i.e. the arrow. It hides it via CSS.
* `cursorWidth`, `cursorColor` – these options refer to the vertical line representing the current time of the audio – you can pass `cursorColor: 'transparent'` to hide this line
* The Cursor plugin – follows the mouse to show another thin line with a timestamp as seen [in this example](https://wavesurfer-js.org/example/cursor/index.html)

I'm assuming you want to hide the mouse pointer, and your configuration seems correct. Except that you don't need to pass a second `hideCursor: true` to the plugin options.

**Reply by redimongo** [reply]
> I think we need to rename some of the "cursor" options because they all refer to different cursors which is confusing:
> 
> 
> 
> * `hideCursor` – this one hides the mouse pointer, i.e. the arrow. It hides it via CSS.
> 
> * `cursorWidth`, `cursorColor` – these options refer to the vertical line representing the current time of the audio – you can pass `cursorColor: 'transparent'` to hide this line
> 
> * The Cursor plugin – follows the mouse to show another thin line with a timestamp as seen [in this example](https://wavesurfer-js.org/example/cursor/index.html)
> 
> 
> 
> I'm assuming you want to hide the mouse pointer, and your configuration seems correct. Except that you don't need to pass a second `hideCursor: true` to the plugin options.

No it was the vertical line I wanted to remove. 

Yes I totally agree it needs to be clearer.

**Reply by katspaugh** [reply]
I see. Yeah, then either `cursorWidth: 0` or `cursorColor: 'transparent'` should do the trick.
